enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,306 @@
+- [/] Test plan is defined
+- [/] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+> 1. Should we manipulate our CSI plugin via the API from the CSI Spec outside the context of K8S Pod provisioning.  For 
+>example, as part of a Operator/Controller co-located with the CSI plugin, can/should we call `DestroyVolme`, `NodeUnstageVolume`, or
+>`NodeUnpublishVolume` when that Operator/Controller detects a change in any watched Secret resources, as part of updating
+>the contents on the host/node?  Or should that Operator/Controller, as part of being hosted as a privileged `DaemonSet`, directly 
+>manipulate the predefined location for the host/node content mapped to the CSI driver?  Sense is the latter is fine,
+>but confirmation from CSI SMEs would be good.
+
+## Summary
+
+Customer feedback on V4.x of OpenShift has included the desire to seamlessly inject certain content (usually of a sensitive nature)
+onto each host/node of a cluster for inclusion in the customer's pod.
+
+At this time, the only means of doing so are 
+
+- Declare `HostPath` based PV/PVCs on the Pod spec, on a per Pod basis, by each author of the Pod
+- Leverage the 4.x machine configuration operator or MCO to update the host file system
+
+There are several drawbacks with these approaches:
+
+- `HostPath` volumes require the privileged bit being set on the Pod, which grants the Pod and Pod author overall permissions
+that cluster administrators will not want, instead of access to just the specific piece of content on the host file system.
+- Host injection via the MCO requires a node reboot for the updates to take effect.  That is generally deemed 
+undesirable for this feature.  
+- The MCO API has not been intuitive for the type of users to date who are interested in introducing and then consuming
+small collections of files on a host.
+
+This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based
+means of OpenShift cluster administrators the means of injecting K8S Secrets onto each host/node in an OpenShift cluster, 
+and then allow owners/authors/users of Pods to opt into the consumption of the content of those Secrets during their 
+Pod's execution.  The means of opting in will entail setting an annotation or label on an object vs. explicitly
+defining the volume related API objects or volume plug points into the Pod.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.
+
+
+## Motivation
+
+Simplify overall usability for injecting and consuming Secret content onto the OpenShift hosts/nodes for use in user
+Pods.
```

Comment:
this document is intended to include enough specifics that someone understands how it will be built/implemented (and can poke holes in that).  So it seems reasonable to have implementation details defined here.  (Whether those details are correct or not, with respect to the right way to implement this CSI driver, is what we are looking for feedback on)

Reply:
The motivation will be changed to `Simplify consumption of Secrets and ConfigMaps defined in one namespace in additional namespaces.` in the next push